prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -530,11 +620,97 @@ def canonicalize_dataset_automatically(
             )  # noqa E501
             return canonicalized_dataset
 
+    def get_datasets_of_required_size(
+        self, datasets_info_dict: dict, prompt_spec: PromptSpec
+    ) -> datasets.DatasetDict:
+        """Combine multiple datasets to get the required size.
+
+        Args:
+            datasets_info_dict (dict): A list of dictionaries representing the datasets.
+            prompt_spec (PromptSpec): An object representing the prompt specification.
```

Comment:
redundant type hints not required
```suggestion
            datasets_info_dict: A list of dictionaries representing the datasets.
            prompt_spec: An object representing the prompt specification.
```